website/static/js/fangorn.js

```diff
@@ -1510,13 +1521,89 @@ var FGItemButtons = {
                         }, 'View'));
                 }
                 if (item.data.permissions && item.data.permissions.edit) {
-                    rowButtons.push(
+                    if (item.data.provider === 'osfstorage') {
+                        if (!item.data.extra.checkout){
+                            rowButtons.push(
+                                m.component(FGButton, {
+                                    onclick: function(event) { _removeEvent.call(tb, event, [item]);  },
+                                    icon: 'fa fa-trash',
+                                    className : 'text-danger'
+                                }, 'Delete'));
+                            rowButtons.push(
+                                m.component(FGButton, {
+                                    onclick: function(event) {
+                                        tb.modal.update(m('', [
+                                            m('p', 'This would mean ' +
+                                                'other contributors cannot edit, delete or upload new versions of this file ' +
+                                                'as long as it is checked-out. You can check it back in at anytime.')
+                                        ]), m('', [
+                                            m('span.btn.btn-default', {onclick: function() {tb.modal.dismiss();}}, 'Cancel'), //jshint ignore:line
+                                            m('span.btn.btn-warning', {onclick: function() {
+                                                $.ajax({
+                                                    method: 'put',
```

Comment:
I'm not sure if I read the code correctly, but I did not see a way to do ``` beforeSend: $osf.setXHRAuthorization, ``` with ``` $osf.putJSON ```. Is there a different way to set this authorization and use ``` osf.putJSON ```